src/pnpq/devices/refactored_switch_thorlabs_osw1310e.py

```diff
@@ -0,0 +1,65 @@
+#
+# Thorlanbs Oprical Switch 1x2 and 2x2 1310E driver
+#       OSW12-1310E & OSW22-1310E
+#
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import serial
+import serial.tools.list_ports
+from serial import Serial
+
+
+class AbstractOpticalSwitchThorlabs1310E(ABC):
+    @abstractmethod
+    def connect(self) -> None:
+        """Open a new connection."""
+
+    def bar_state(self) -> None:
+        """Set the switch to the bar state."""
+
+    def cross_state(self) -> None:
+        """Set the switch to the cross state."""

```

Comment:
All three methods must be decorated with `@abstractmethod`.

```suggestion
    @abstractmethod
    def connect(self) -> None:
        """Open a new connection."""

    @abstractmethod
    def bar_state(self) -> None:
        """Set the switch to the bar state."""

    @abstractmethod
    def cross_state(self) -> None:
        """Set the switch to the cross state."""
```